pkg/cmd/extension/command.go

```diff
@@ -305,7 +333,11 @@ func NewCmdExtension(f *cmdutil.Factory) *cobra.Command {
 						return err
 					}
 
-					if err := checkValidExtension(cmd.Root(), m, repo.RepoName()); err != nil {
+					if ext, err := checkValidExtension(cmd.Root(), m, repo.RepoName()); err != nil {
+						if forceFlag && ext != nil {
+							return upgradeFunc(ext.Name(), forceFlag, false)
```

Comment:
I think the error matching needs to be more precise here. For instance, the error can be that the extension repo doesn't start with `gh-` or that, or that the extension name matches a built-in command name. In both cases, `--force` should not take effect.

Reply:
Right, but in those cases the `ext` return is `nil`, hence the check on `ext` and not `err`. If it helps, I will add a comment better explaining that. I had considered some refactoring of `checkValidExtension` initially to separate out just that logic, but it ends up duplicate a bit of code no matter what so changing the return of `checkValidExtension` to return the already-parsed `Extension` for which I need the `Name` anyway made the most sense.